test/Dialect/Moore/basic.mlir

```diff
@@ -302,6 +302,11 @@ moore.module @Expressions(
   // CHECK: moore.struct_inject [[STRUCT1]], "a", [[B]] : struct<{a: i32, b: i32}>, i32
   moore.struct_inject %struct1, "a", %b : struct<{a: i32, b: i32}>, i32
 
+  // CHECK: moore.string_constant "Test" : i128
+  moore.string_constant "Test" : i128
+  // CHECK: moore.string_constant "" : i128
+  moore.string_constant "" : i128
+
```

Comment:
Thanks for adding these! Can you also add a similar thing to the MooreToCore test, to check that the lowering works?

Reply:
Thank you for pointing out the correct location of the `MooreToCore` test. I've realized that my `StringConstantOpConv` conversion currently ignores the type of the resulting value, which shouldn’t be the case. For example:

```mlir
%str = moore.string_constant "Test" : i2
```

is currently converted to:

```mlir
%c1415934836_i32 = hw.constant 1415934836 : i32
```

when compiled with `circt-opt` using the `--convert-moore-to-core` flag. I'll update the conversion to ensure the resulting type is set correctly.

PD: In this example, I'm uncertain about the expected outcome: should it convert to `hw.constant 1 : i2`, or should it fail to convert?

Reply:
Good question. Either should work. It might be a bit better to fail the conversion. The string being larger than the type should be caught in the frontend already, so the user should never see this error. But erroring out is good to catch compiler bugs 🙂

Reply:
Thank @jpinot for your work on `StringConstantOpConv` :partying_face:! @fabianschuiki is all right! Slang can guarantee the correct type width. And You're able to don't worry about the type conversion. Because @maerhart has already handled the related error on https://github.com/llvm/circt/pull/7631. I think maybe you should tweak your test from 
```
%str = moore.string_constant "Test" : i2
%str_empty = moore.string_constant "" : i128
``` 
to 
```
%str = moore.string_constant "Test" : i32
%str_empty = moore.string_constant "" : i8
```

Reply:
Thank you, @fabianschuiki and @hailongSun2000, for taking the time to review my PR. I've made the following updates based on your feedback:

- Adjusted the test to match type with string size.
- Added checks to validate that the type and type size are smaller or equal than the string size.
- Included a new test case to verify this functionality.

Additionally, I noticed an issue with the run command in the test file. I corrected it in a [separate commit](https://github.com/llvm/circt/pull/7752/commits/8aa4794e254680e081daee47c510a91789cf452a). Please let me know if I misunderstood  or if you prefer this change to be in a separate PR.

Reply:
You're right! Thanks for catching this detail.